src/components/data/TableHead.js

```diff
@@ -0,0 +1,55 @@
+import classnames from 'classnames';
+import { useContext } from 'preact/hooks';
+
+import { downcastRef } from '../../util/typing';
+
+import TableContext from './TableContext';
+import TableSectionContext from './TableSectionContext';
+
+/**
+ * @typedef {import('../../types').PresentationalProps} CommonProps
+ * @typedef {Omit<import('preact').JSX.HTMLAttributes<HTMLElement>, 'size'>} HTMLAttributes
+ * @typedef {import('./TableSectionContext').TableSection} TableSection
+ */
+
+/**
+ * Render a table head section
+ *
+ * @param {CommonProps & HTMLAttributes} props
+ */
+const TableHeadNext = function TableHead({
+  children,
+  classes,
+  elementRef,
+
+  ...htmlAttributes
+}) {
+  const tableContext = useContext(TableContext);
+
+  const sectionContext = /** @type {TableSection} */ ({
+    section: 'head',
+  });
+
+  return (
+    <TableSectionContext.Provider value={sectionContext}>
+      <thead
+        {...htmlAttributes}
+        ref={downcastRef(elementRef)}
+        className={classnames(
+          // Top border is set here instead of on `Table` to prevent sticky
+          // headers from causing a border wobble on scroll
+          'bg-grey-2 border-t',
```

Comment:
There was a problem with borders, but the lack of a border-color class is a red herring. There's a default border color set in the shared tailwind preset:

```
      borderColor: {
        DEFAULT: '#dbdbdb',
      },
```

(Default width and style is `1px` and `solid` out of the box).

This is both an ergonomic affordance — in most contexts[^1] you can just use `border` and get a 1-pixel solid border in the default color — and a consistency lock (borders are always the same color unless you expressly change the border color).

[^1]: "Most contexts" because this doesn't work consistently after certain resets we apply within shadow DOMs. I've called it out with comments when that is the case, e.g. in the client's `AdderToolbar`:

```
        // Because `.AdderToolbar` rules reset `all:initial`, we cannot use
        // default border values from Tailwind and have to be explicit about
        // border attributes
        'border border-solid border-grey-3',
```